File: Lua/002.lua
function even_fib(limit)
    local a, b, sum = 0, 1, 0
    while (b < limit) do
        local temp = a + b
        if (temp > limit) then
            break
        end

        a = b
        b = temp
        if (temp % 2 == 0) then
            sum = sum + temp
        end
    end

    return sum
end

print(even_fib(4000000))  -- 4613732

File: Go/002.go
package main

import "fmt"

func main() {
	f := fib()
	var sum int = 0
	for {
		a := f()
		if a%2 == 0 {
			sum += a
		}
		if a >= 4000000 {
			break
		}
	}
	fmt.Println(sum)
}

func fib() func() int {
	a, b := 0, 1
	return func() int {
		a, b = b, a+b
		return a
	}
}
